client/src/lib/data.ts

```diff
@@ -13,3 +17,174 @@ export function readToken(): string {
   if (!token) throw new Error('No token found');
   return token;
 }
+
+export async function postSignUp(event: FormEvent<HTMLFormElement>) {
+  const formData = new FormData(event.currentTarget);
+  const userData = Object.fromEntries(formData);
+  const req = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(userData),
+  };
+  const res = await fetch('/api/auth/sign-up', req);
+  if (!res.ok) {
+    throw new Error(`Fetch error: ${res.status}`);
+  }
+  const user = await res.json();
+  return user;
+}
+
+export async function verifySignIn(event: FormEvent<HTMLFormElement>) {
+  const formData = new FormData(event.currentTarget);
+  const userData = Object.fromEntries(formData);
+  const req = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(userData),
+  };
+  const resp = await fetch('/api/auth/sign-in', req);
+  if (!resp.ok) {
+    throw new Error(`${resp.status}`);
+  }
+  const payload = await resp.json();
+  return payload;
+}
+
+export async function getDetails(filmId: string | undefined) {
```

Comment:
For historical reasons (actually, it comes from Java, in particular an old technology named Java Beans), the term "get" is usually associated with functions that are synchronous, quick, and do almost nothing. `read`, `fetch`, `load`, etc., would be a better prefix than `get`.